### ISSUE TITLE
bzlmod: with 6.3.0 generated lockfile it changes when running projects on different platforms if using `requirements_darwin` and `requirements_windows` parameters`

### ISSUE THREAD

**Issue by aignas** [bazelbuild/rules_python#1345]
# 🐞 bug report

### Affected Rule

`bzlmod` `pip.parse` extension.

### Is this a regression?

No, this is not a regression.

### Description

`rules_python` repository rules result in a different dependency closure depending on which platform the rules are being run on. This means that the users may be unable to add the `MODULE.bazel.lock` to their version control if multiple platforms are being used.

## 🔬 Minimal Reproduction

```starlark
pip.parse(
    hub_name = "publish_deps",
    python_version = "3.10",
    requirements_darwin = "@rules_python//tools/publish:requirements_darwin.txt",
    requirements_lock = "@rules_python//tools/publish:requirements.txt",
    requirements_windows = "@rules_python//tools/publish:requirements_windows.txt",
)
```

Then creating a lock with `bazel 6.3.0` on Linux and a different platform will yield to changes in the lock file.

## 🔥 Exception or Error

The lock file differs on Linux and Mac.

## 🌍  Your Environment

* Linux Ubuntu 20.04 LTS
* Latest MacOS on Intel Mac

**Output of `bazel version`:** `6.3.0`

**Rules_python version:** `0.24.0`

**Comment by Wyverald** [comment]
/sub. also cc @meteorcloudy @SalmaSamy

**Comment by Wyverald** [comment]
This is not necessarily something rules_python can solve. But then again, this is not necessarily something we can solve at all. @aherrmann will remember the lengthy discussion we had on conditional dependencies and how that might interact with lockfiles. If users need to produce different repos on different platforms, they will necessarily get different lockfiles; one way to deal with that might be to manually multiplex between different lockfiles themselves. We may also need to provide a flag to choose the lockfile path (so the user might say `--lockfile_name=MODULE.bazel.lock.darwin`).

**Comment by meteorcloudy** [comment]
> We may also need to provide a flag to choose the lockfile path (so the user might say --lockfile_name=MODULE.bazel.lock.darwin).

Yeah, combining with [--enable_platform_specific_config](https://bazel.build/reference/command-line-reference#flag--enable_platform_specific_config), this could be a workaround for major platforms.

**Comment by aignas** [comment]
@Wyverald, @meteorcloudy, right now I don't see a way to switch the platform config per `(os, arch)` tuple (e.g. Having a different lock file for `macos_x86_64` and `macos_arm64` and `linux_x86_64` and `linux_arm64`. What would be the idiomatic way to handle such cases?

**Comment by meteorcloudy** [comment]
[--enable_platform_specific_config](https://bazel.build/reference/command-line-reference#flag--enable_platform_specific_config) doesn't work for different arches, we need to come up with a better solution for the lockfile to work with multiple platforms.
